parallel-install/pkg/deployment/core.go

```diff
@@ -51,7 +53,13 @@ func newCore(cfg *config.Config, ob *OverridesBuilder, kubeClient kubernetes.Int
 		removeFromComponentList(clList, incompatibleLocalComponents)
 	}
 
-	registerOverridesInterceptors(ob)
+	commonRetryOpts := []retry.Option{
+		retry.Delay(time.Duration(cfg.BackoffInitialIntervalSeconds) * time.Second),
+		retry.Attempts(uint(cfg.BackoffMaxElapsedTimeSeconds / cfg.BackoffInitialIntervalSeconds)),
```

Comment:
I have mixed feelings about this... These config params supports `https://github.com/cenkalti/backoff` library, which has a more "declarative" approach to retries. The default values are 30 and 1, meaning your code performs 30 retries - a bit too much IMO. In addition, `BackoffInitialIntervalSeconds` can be zero, which would lead to panic here. I would consider reasonable constant values here, like 3 retries with 2 seconds delay. Another approach is to use `BackoffMaxElapsedTimeSeconds` and divide it by a constant number of retries (like 3 or 5) to get the fixed delay value.